### PR TITLE
Fix #98: TonY Portal should enforce retention on history files

### LINKED ISSUE #98: TonY Portal should enforce retention on history files
Currently, history files are retained forever. The retention period should be configurable and TonY Portal should take care of enforcing retention.

As part of retention, we can also clean up in-progress files that are older than the retention period. (These are probably jobs that crashed or encountered other abnormal conditions.)

Internal Jira: LIHADOOP-43855

### PR DESCRIPTION
**Pull request by erwa**
* Added HistoryFileCleaner thread
* Added configs `tony.history.retention-sec` (default 30 days) and `tony.history.cleaner-interval-ms` (default 6 hours)
* Added unit tests
* Minor tweaks to startTonyPortal.sh
* Tested in real cluster

### CODE REVIEW

File: tony-core/src/main/resources/tony-default.xml [hunk 1]
```diff
@@ -261,6 +261,18 @@
     finished directory.</description>
   </property>
 
+  <property>
+    <name>tony.history.retention-sec</name>
+    <value>2592000</value>
+    <description>How many seconds to retain history files for.</description>
```
Comment (oliverhu) [inline tony-core/src/main/resources/tony-default.xml]: can you put how many days is 2592000 in the desc? same below

File: tony-portal/app/history/HistoryFileCleaner.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import javax.inject.Singleton;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import play.Logger;
+import utils.ConfigUtils;
+
+
+@Singleton
+public class HistoryFileCleaner {
```
Comment (oliverhu) [inline tony-portal/app/history/HistoryFileCleaner.java]: HistoryFilePurger sounds more professional ?

File: tony-portal/app/history/HistoryFileCleaner.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        LOG.error("Encountered exception while cleaning history directories", e);
+      }
+    }, 0, cleanerIntervalMs, TimeUnit.MILLISECONDS);
+  }
+
+  /**
+   * Deletes all year/month/day directories in the finished dir prior to the cutoff date.
+   */
+  @VisibleForTesting
+  static void cleanFinishedDir(FileSystem fs, Path finishedDir, LocalDate cutOffDate) throws IOException {
```
Comment (oliverhu) [inline tony-portal/app/history/HistoryFileCleaner.java]: why is cleanFinishedDir so different from cleanIntermediateDir ?

File: tony-portal/app/history/HistoryFileCleaner.java (hunk 2 above)
Comment (erwa) [reply tony-portal/app/history/HistoryFileCleaner.java]: Agreed.

File: tony-portal/app/history/HistoryFileCleaner.java (hunk 3 above)
Comment (erwa) [reply tony-portal/app/history/HistoryFileCleaner.java]: The structure of finished is yyyy/mm/dd/<app_id> whereas intermediate directly contains <app_id> folders.

For purging finished, we have to parse yyyy/mm/dd and interpret as a date, whereas for intermediate, I'm just using the last modification time of the <app_id> directory as a heuristic.

File: tony-core/src/main/resources/tony-default.xml (hunk 1 above)
Comment (erwa) [reply tony-core/src/main/resources/tony-default.xml]: Good idea. Updated.

**Pull request merged**